src/main/java/org/apache/maven/reporting/MavenReport.java

```diff
@@ -53,9 +53,10 @@ public interface MavenReport {
     void generate(Sink sink, Locale locale) throws MavenReportException;
 
     /**
-     * Get the base name used to create report's output file(s).
+     * Get the path relative to {@link #getReportOutputDirectory()} where the report's main output
+     * file will be written. The last component is expected to be a basename only.
```

Comment:
still not sure what "basename" means. I guess the point is it's a directory, not a filename?

Reply:
Since all reports are output format agostic, the output path does not contain any extension, this is handled by a sink. Basename of the output file. E.g., index, foo, etc. How would you word it?

Reply:
OK, so I indeed did not understand what "basename" meant in this context, which is de facto evidence that this isn't clear enough. Either define "basename" as the "filename without an extension or period" or perhaps write something like, "The last component is the name of the file without any extension. The extension (.pdf, .html, etc.) will be added by the sink. For example, repodocs/subtext/analysis_report"

though perhaps this is hard to explain because the API itself is weird. I rarely encounter a path to a file that includes directories and file anme but not file extension.

Reply:
> OK, so I indeed did not understand what "basename" meant in this context, which is de facto evidence that this isn't clear enough. Either define "basename" as the "filename without an extension or period" or perhaps write something like, "The last component is the name of the file without any extension. The extension (.pdf, .html, etc.) will be added by the sink. For example, repodocs/subtext/analysis_report"
> 
> though perhaps this is hard to explain because the API itself is weird. I rarely encounter a path to a file that includes directories and file anme but not file extension.

That's exactly what I thought about. The issue is that the plugin does not know the sink impl therefore it shall not make any assumptions. It could also be a sink to an output stream of a servlet.

Reply:
@elharo Done, please have another look.